04 pig latin

AdrianArcas opened the pull request.

AdrianArcas committed:
Solve 04_pig_latin

AdrianArcas committed:
Add test-driving bonus

AdrianArcas committed:
Generalize counting groups of letters phonemes

AdrianArcas committed:
Refactor loops

salcudeann commented on 04_pig_latin/pig_latin.rb:
```diff
@@ -0,0 +1,80 @@
+def translate(sentence)
+
+  sentence.split.map { |word|
+    cap = false
```
Could you extract all these logic rules to a separate method and apply the method for every word?  e.g. `translate(word)` to return the formatted word

AdrianArcas committed:
Move logic rules from translate to translate_word and refactor

AdrianArcas committed:
Write new function to call on each word

Created a new function "translate_word" to be called on each word to
improve readability using existing logic
Found a way to get rid of the infinite loop, refactored vowel_outside_phonemes to
outside_phonemes? and refactored some conditions for readability
